kratos/geometries/point.h

```
@@ -195,16 +197,138 @@ class Point : public array_1d<double, 3>
+        return this->Z();
+    }
+
+    /**
+     * @brief This method returns the X initial coordinates (constant)
+     * @return X initial coordinates
+     * @note Must be override in tehd erived classes with initial position
+     */
+    virtual double X0() const
+    {
+        KRATOS_WARNING("Point") << "This class does not have InitialPosition coordinates. Use an derived class and override this method in order to do so" << std::endl;
+        return this->X();
+    }
+
+    /**
+     * @brief This method returns the Y initial coordinates (constant)
+     * @return Y initial coordinates
+     * @note Must be override in tehd erived classes with initial position
+     */
+    virtual double Y0() const
+    {
+        KRATOS_WARNING("Point") << "This class does not have InitialPosition coordinates. Use an derived class and override this method in order to do so" << std::endl;
+        return this->Y();
+    }
+
+    /**
+     * @brief This method returns the Z initial coordinates (constant)
+     * @return Z initial coordinates
+     * @note Must be override in tehd erived classes with initial position
+     */
+    virtual double Z0() const
+    {
+        KRATOS_WARNING("Point") << "This class does not have InitialPosition coordinates. Use an derived class and override this method in order to do so" << std::endl;
+        return this->Z();
+    }
+
+    /**
+     * @brief This method sets the initial coordinates
+     * @param rNewInitialPosition The new coordinates
+     * @note Must be override in tehd erived classes with initial position
+     */
+    virtual void SetInitialPosition(const PointType& rNewInitialPosition)
+    {
+        KRATOS_WARNING("Point") << "This class does not have InitialPosition coordinates. Use an derived class and override this method in order to do so" << std::endl;
```

I'd throw an error in this case.

---

Done